### PR TITLE
Added a check for duplicate keys (fixes #536)

### LINKED ISSUE #536: Config::sanityCheck() bug
Small bug I encountered today, _sanityCheck_ doesn't catch the following case:

```
- !<Look>
  name: primary_dn
  process_space: log
  transform: !<FileTransform> {src: my_file.cdl, interpolation: linear}
  transform: !<FileTransform> {src: my_file.cdl, interpolation: linear}
```

Where the YAML "transform" tag is defined twice, behaviour was that it didn't error it just grabbed the last defined "transform" and used that FileTransform.

### PR DESCRIPTION
**Pull request by rashil2000**
Signed-off-by: Rashil Gandhi <rashil2000@gmail.com>

### CODE REVIEW

File: src/OpenColorIO/OCIOYaml.cpp [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include <cstring>
 
 #include <OpenColorIO/OpenColorIO.h>
+#include <unordered_set>
```
Comment (hodoulp) [inline src/OpenColorIO/OCIOYaml.cpp]: We order all the includes, and also regroup them based on 'family'. So, you should move that one up to line 5 (i.e. keep the empty line with the existing include at line 6).

File: src/OpenColorIO/OCIOYaml.cpp [hunk 2]
```diff
@@ -270,6 +271,29 @@ inline void throwValueError(const std::string & nodeName,
     throw Exception(os.str().c_str());
 }
 
+// Duplicate Checker
+
+inline void checkDuplicates(const YAML::Node& node)
```
Comment (hodoulp) [inline src/OpenColorIO/OCIOYaml.cpp]: The global methods must start with an upper case letter (i.e. we did not revisit all the existing files to conform them to the coding style) and add a space between `YAML::Node` and `&`.

File: src/OpenColorIO/OCIOYaml.cpp [hunk 3]
```diff
@@ -270,6 +271,29 @@ inline void throwValueError(const std::string & nodeName,
...
+{
+    std::string key;
+    std::unordered_set<std::string> keyset;
+
+    for (const auto & iter : node)
+    {
+        const YAML::Node& first = iter.first;
```
Comment (hodoulp) [inline src/OpenColorIO/OCIOYaml.cpp]: Same comment than the one at line 276.

File: tests/cpu/Config_tests.cpp [hunk 4]
```diff
@@ -181,6 +181,36 @@ OCIO_ADD_TEST(Config, simple_config)
...
+        "    name: raw_duplicated\n"
+        "    name: raw\n"
+        "\n";
+
+    std::istringstream is;
+    is.str(SIMPLE_PROFILE);
+    OCIO::ConstConfigRcPtr config;
+    OCIO_CHECK_THROW_WHAT(OCIO::Config::CreateFromStream(is),
+                              OCIO::Exception,
+                              "Key-value pair with key 'name' "
```
Comment (hodoulp) [inline tests/cpu/Config_tests.cpp]: When the argument list is on several lines, they must be aligned, and the max. line length is 100 characters. You should adjust the code accordingly.

File: tests/cpu/Config_tests.cpp [hunk 5]
```diff
@@ -181,6 +181,36 @@ OCIO_ADD_TEST(Config, simple_config)
     OCIO_CHECK_NO_THROW(config->sanityCheck());
 }
 
+OCIO_ADD_TEST(Config, simple_config_with_duplicates)
+{
+
+    constexpr char SIMPLE_PROFILE[] =
+        "ocio_profile_version: 2\n"
+        "search_path: \n"
```
Comment (hodoulp) [inline tests/cpu/Config_tests.cpp]: I missed than one, please add `luts` as the directory where to find the LUT files. 
The line must be:  `"search_path: luts\n"`.

File: src/OpenColorIO/OCIOYaml.cpp (hunk 2 above)
Comment (rashil2000) [reply src/OpenColorIO/OCIOYaml.cpp]: I had a confusion regarding this too. I tried to visited all the functions on `OCIOYaml.cpp` to check for both the capitalization and the space before the '&' symbol. Some functions had the space, others didn't. So I made a guess.

Comment (hodoulp) [reply src/OpenColorIO/OCIOYaml.cpp]: That's the coding style but not all files are following it. So, only modifying the code you changed is enough.

File: src/OpenColorIO/OCIOYaml.cpp (hunk 1 above)
Comment (hodoulp) [reply src/OpenColorIO/OCIOYaml.cpp]: To clarify the point:
```
#include <cstring>
#include <unordered_set>

#include <OpenColorIO/OpenColorIO.h>
```

File: src/OpenColorIO/OCIOYaml.cpp [hunk 6]
```diff
@@ -270,6 +270,29 @@ inline void throwValueError(const std::string & nodeName,
...
+{
+    std::string key;
+    std::unordered_set<std::string> keyset;
+
+    for (const auto & iter : node)
+    {
+        const YAML::Node & first = iter.first;
+        load(first, key);
+        if (keyset.find(key) == keyset.end())
+            keyset.insert(key);
```
Comment (hodoulp) [inline src/OpenColorIO/OCIOYaml.cpp]: Even a one-line branch needs to be surrounded by curly brackets.

**Pull request merged**